pkg/storage/bucket/s3/config.go

```diff
@@ -78,6 +78,16 @@ type HTTPConfig struct {
 
 	// Allow upstream callers to inject a round tripper
 	Transport http.RoundTripper `yaml:"-"`
+
+	TLSConfig TLSConfig `yaml:"tls_config"`
+}
+
+// TLSConfig configures the options for TLS connections.
```

Comment:
Elsewhere in Mimir, we use `tls.ClientConfig` from dskit. Do you mind changing to that struct for consistency with existing TLS config? There's an example [here](https://github.com/grafana/mimir/blob/52e7125f747c3d992491a7b52642abc97c4f074d/pkg/querier/store_gateway_client.go#L104-L107). When you do, it probably makes sense to have the YAML and CLI flags be inline like the other places it's used: 
```
`yaml:",inline"`
```

Reply:
A few notes about this request that I have difficulties understanding:

Firstly, I see that `HTTPConfig` already has some of the configurations present on the `tls.ClientConfig` from dskit:
https://github.com/grafana/mimir/blob/52e7125f747c3d992491a7b52642abc97c4f074d/pkg/storage/bucket/s3/config.go#L72
I believe that enabling this as a configuration option in both places is not the correct way to go.

Secondly, the original suggestion was to allow only relevant values to be passed to the exthttp client as `exthttp.TLSConfig` doesn't have some of the options allowed with `tls.ClientConfig` from dskit (such as `MinVersion`) which I don't really know how to handle.

I apologize if my wording sounds a bit rude, I'm quite new to Mimir's codebase so I might not understand this correctly.

Reply:
Ah, so it does. In that case, could you adjust the struct a bit so that the settings being added match names if we were using `tls.ClientConfig`?

E.g. make `TLSConfig` inline with `HTTPConfig` , name the fields the same like `tls_ca_path`, and name the CLI flags like `-ruler-storage.s3.http.tls-ca-path`, etc

Reply:
I'll do so, please note that in other components referance `insecure_skip_verify`  as `tls_insecure_skip_verify` which would cause some more inconsistencies later on.

Reply:
Another issue that shows in the tests, this PR will change `<prefix>.s3.http.tls.<option>` to `<prefix>.s3.http.tls-<options>`. Sounds like a big change to me, are there any guidelines on changing configuration parameters naming to this degree?

Reply:
> I'll do so, please note that other components reference insecure_skip_verify as tls_insecure_skip_verify which would cause some more inconsistencies later on.

That's fine to leave as-is. We can deprecate the old option in another PR or just leave it.

> Another issue that shows in the tests, this PR will change <prefix>.s3.http.tls.<option> to <prefix>.s3.http.tls-<options>. Currently I resolved it by running make reference-help but it sounds like a big change to me, are there any guidelines on changing configuration parameters naming to this degree?

It's expected to need to run `make reference-help` after changing CLI flags. These are all new options so there shouldn't be any issue changing them during the course of this PR.